src/kv/mod.rs

```diff
@@ -162,7 +172,10 @@ impl Kv {
     pub async fn put(&mut self, req: EtcdPutRequest) -> Res<EtcdPutResponse> {
         let key = req.get_key();
         let value = req.get_value();
-        let resp = self.client.put_async(&req.into())?.await?;
+        let resp = retryable!(|| async {
+            let resp = self.client.put_async(&req.clone().into())?;
```

Comment:
Is it possible to use move async {}  to avoid clone()?

Reply:
Some for other places.

Reply:
I tried `async move`, but the error is:

`move occurs because `self` has type `&mut auth::Auth`, which does not implement the `Copy` trait`
`move occurs because `req` has type `auth::authenticate::EtcdAuthenticateRequest`, which does not implement the `Copy` trait`

Not sure if we should add `Copy` trait to those fields or just use `clone` here.